src/contact.rs

```diff
@@ -291,29 +292,28 @@ impl Contact {
         context: &Context,
         addr: impl AsRef<str>,
         min_origin: Origin,
-    ) -> u32 {
+    ) -> Result<Option<u32>> {
         if addr.as_ref().is_empty() {
-            return 0;
+            return Err(Error::msg("lookup_id_by_addr: empty address"));
```

Comment:
Why not `bail!`?

Reply:
I guess I'm just not a big macro fan, for me this is more readable (I never even thought of `bail!` to be honest).  If we prefer the `bail!` style I could change it though.

Reply:
We have `bail!` everywhere but no `Error::msg`. I even had to double check it's the same thing, so I think it's better to change to `bail!` for consistency.